### PR TITLE
doc: frontend doc rework

### PR DESCRIPTION
**Pull request by davecoates**
- Add documentation for templatetags in page so we have full control over formatting. The autofunction solution doesn't work well for templatetags as it can't properly extract any arguments
- Add an overview page
- Put all API style docs under the API section
- Enhance the alliance_ui docs
- Many other minor fixes and enhancements

Generated docs are [here](https://alliance-platform--30.org.readthedocs.build/en/30/index.html)

### CODE REVIEW

File: packages/ap-frontend/docs/overview.rst [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+- Each component rendered in a template queues itself to be rendered with :meth:`~alliance_platform.frontend.bundler.context.BundlerAssetContext.queue_ssr`.
+- :class:`~alliance_platform.frontend.bundler.middleware.BundlerAssetContextMiddleware` accesses the context and retrieves all the queued SSR items.
+- It then serializes the queued items and calls out to javascript, which renders each component to static HTML and returns it
+
+    - In development this is handled by ``dev-server.ts``. This allows Vite to process the required modules without having to do a full production build
+    - In production it is handled by ``production-ssr-server.ts`` which works with the production built files.
+
+Currently, the only thing that gets rendered on the server is React components. :class:`~alliance_platform.frontend.templatetags.react.ComponentSSRItem`
+is used to describe the component that needs to be rendered. See it's documentation for details on how each component
```
Comment (micahklaz) [inline packages/ap-frontend/docs/overview.rst:133]: Typo: it's -> its

File: packages/ap-frontend/docs/overview.rst [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    {% load react %}
+
+    <!-- Default export -->
+    {% component "components/Button" type="primary" %}
+        <strong>Click Me</strong>
+    {% endcomponent %}
+    <!-- Named export -->
+    {% component "components/Table" "Column" %}Header{% endcomponent %}
+    <!-- Pass a comopnent as a prop to another component -->
+    {% component "components/Icons" "Menu" as menu_icon %}
```
Comment (micahklaz) [inline packages/ap-frontend/docs/overview.rst:163]: Does this need an `endcomponent` tag?

File: packages/ap-frontend/docs/templatetags.rst [hunk 3]
```diff
@@ -1,22 +1,665 @@
...
+    {% bundler_preamble %}
+
+.. templatetag:: bundler_dev_checks
+
+``bundler_dev_checks``
+----------------------
+
+Performs dev specific checks and may render some HTML to communicate messages to user
+
+Currently check if the dev server is running for this project, and if not displays an error.
```
Comment (micahklaz) [inline packages/ap-frontend/docs/templatetags.rst:204]: Typo: check -> checks

File: packages/ap-frontend/docs/templatetags.rst [hunk 4]
```diff
@@ -1,22 +1,665 @@
...
+Alternatively, you can disable SSR entirely by passing ``disable_ssr=True`` to :class:`~alliance_platform.frontend.bundler.vite.ViteBundler`.
+
+Options
+~~~~~~~
+
+Various options can be passed to the component tag. To differentiate from actual props to the component they are
+prefixed with `ssr:` for server side rendering options, `component:` for general component options, or `container:`
+for options relating to the container the component is rendered into.
+
+- ``ssr:disabled=True`` - if specified, no server side rendering will occur for this component
```
Comment (micahklaz) [inline packages/ap-frontend/docs/templatetags.rst:426]: If rendering on the client, will this automatically handle `<Suspense>` and displaying some sort of loading icon? If not, is that something we want to support (perhaps as an additional `ssr` argument), or should mention in the documentation?

**Review by micahklaz — CHANGES_REQUESTED** [review]
Couple typos and a question

File: packages/ap-frontend/docs/overview.rst (hunk 2 above)
Comment (davecoates) [reply packages/ap-frontend/docs/overview.rst:163]: yep, added

File: packages/ap-frontend/docs/templatetags.rst (hunk 4 above)
Comment (davecoates) [reply packages/ap-frontend/docs/templatetags.rst:426]: Suspense isn't applicable here as it's not loading it within a React tree. We could support an option for it, e.g. `ssr:placeholder=...` to allow rendering something.

I'll add a note about it when SSR is disabled

Comment (davecoates) [reply packages/ap-frontend/docs/templatetags.rst:429]: https://github.com/AllianceSoftware/alliance-platform-py/issues/31 tracks this

**Pull request merged**